### ISSUE TITLE
Latest Conda doesn't work against Artifactory with "conda install"

### ISSUE THREAD

**Issue by yuvalreches** [conda/conda#8623]
## Current Behavior
The latest Conda client (tested with 4.6.14) fails to perform `conda install X` against JFrog Artifactory.
Any version of 4.5.x and below does work.

Seems to be an issue with a new commit on `conda/common/url.py` line 119, in `path_to_url`

### Steps to Reproduce
1. Run Artifactory (https://jfrog.com/download-artifactory-pro/)
2. Set up Conda repositories (top right Admin menu --> Quick setup --> Conda)
3. Install the latest Conda client
4. Configure Conda client to work against Artifactory: 
Go to Artifctory Conda repo, click on `Set Me Up` in the top right
under ~/.condarc add the text from the Set Me Up dialog
(starts with: `channel_alias: http://admin:<KEY>@localhost:8081/artifactory/api/conda/conda/......`)
5. Perform `conda install scipy`
6. You will notice a `IndexError: string index out of range` of the client.

The following stacktrace appears:
```
    Traceback (most recent call last):
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/exceptions.py", line 1003, in __call__
        return func(*args, **kwargs)
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/cli/main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/cli/conda_argparse.py", line 82, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/cli/main_install.py", line 20, in execute
        install(args, parser, 'install')
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/cli/install.py", line 275, in install
        handle_txn(unlink_link_transaction, prefix, args, newenv)
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/cli/install.py", line 304, in handle_txn
        unlink_link_transaction.execute()
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/core/link.py", line 236, in execute
        self.verify()
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/core/link.py", line 213, in verify
        self.prepare()
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/core/link.py", line 205, in prepare
        stp.remove_specs, stp.update_specs)
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/core/link.py", line 280, in _prepare
        for prec, pcrec in zip(link_precs, pkg_cache_recs_to_link))
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/core/link.py", line 280, in <genexpr>
        for prec, pcrec in zip(link_precs, pkg_cache_recs_to_link))
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/gateways/disk/read.py", line 104, in read_package_info
        channel=Channel(record.schannel or record.channel),
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/models/channel.py", line 37, in __call__
        c = Channel._cache_[value] = Channel.from_value(value)
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/models/channel.py", line 106, in from_value
        return Channel.from_url(path_to_url(value))
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/_vendor/auxlib/decorators.py", line 59, in _memoized_func
        result = func(*args, **kwargs)
      File "/Users/yuvalr/miniconda2/lib/python2.7/site-packages/conda/common/url.py", line 119, in path_to_url
        if path[1] == ':':
    IndexError: string index out of range
```

I would mention that I noticed the `url.py` code has been changed in the latest version.
During the run, the `path` variable gets the value of `/`.
However the new code looks for ":" at second char for some reason.

On the old version (4.5.12) I see the value of path is also `/`, and it works perfectly with Artifactory.

Stacktrace shows problem is here:
```
conda/common/url.py", line 119, in path_to_url
        if path[1] == ':':
    IndexError: string index out of range
```

But I couldn't figure out where the path is derived from.
Seems to me the condition should be different.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>
```
     active environment : None
       user config file : /Users/yuvalr/.condarc
 populated config files : /Users/yuvalr/.condarc
          conda version : 4.6.14
    conda-build version : not installed
         python version : 2.7.16.final.0
       base environment : /Users/yuvalr/miniconda2  (writable)
           channel URLs : http://admin:<KEY>@localhost:8080/artifactory/api/conda/conda-remote/osx-64
                          http://admin:<KEY>@localhost:8080/artifactory/api/conda/conda-remote/noarch
          package cache : /Users/yuvalr/miniconda2/pkgs
                          /Users/yuvalr/.conda/pkgs
       envs directories : /Users/yuvalr/miniconda2/envs
                          /Users/yuvalr/.conda/envs
               platform : osx-64
             user-agent : conda/4.6.14 requests/2.21.0 CPython/2.7.16 Darwin/18.5.0 OSX/10.14.4
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>
```
==> /Users/yuvalr/.condarc <==
channels:
  - http://admin:<KEY>@localhost:8080/artifactory/api/conda/conda-remote/
default_channels:
  - http://admin:<KEY>@localhost:8080/artifactory/api/conda/conda-remote/
channel_alias: http://admin:<KEY>@localhost:8080/artifactory/api/conda/conda-remote/
```
</p></details>

**Comment by msarahan** [comment]
Artifactory is a 3rd party index, and while conda doesn't intentionally break it, we also make no effort to ensure that it will not break, nor do we "fix" conda when it breaks.

**Comment by AlexMikhalev** [comment]
Bump. Same issue.
@msarahan if you can propose workaround it would be more helpful.

**Comment by AlexMikhalev** [comment]
The actual fix has already been committed, 
https://github.com/conda/conda/commit/b030abfc961d9de7c351f029425d58280606d995 
the problem is current miniconda.sh and anaconda.sh shipped with broken get_url which means you would not be able to install updated conda package. 
Solution: open the editor and fix manually
`
if len(path) > 1 and path[1] == ':':
`

**Comment by tpanza** [comment]
Any insight on when we can see an update to conda 4.6.x on Linux that applies this fix?

**Comment by rgommers** [comment]
This can be closed, the solution landed in conda 4.7.0
